src/main/java/pl/javastart/zadanie20/UserController.java

```diff
@@ -0,0 +1,53 @@
+package pl.javastart.zadanie20;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+
+@Controller
+public class UserController {
+
+    private final UserRepository userRepository;
+
+    @Autowired
+    public UserController(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    @RequestMapping("/users")
+    @ResponseBody
+    public String getAddedUsers() {
+        StringBuilder result = new StringBuilder();
+        List<User> users = userRepository.getUsers();
+        for (User user : users) {
+            result.append(user.toString()).append("<br/>");
+        }
+        return result.toString();
+    }
+
+    @RequestMapping("/add")
+    public String addUserForm(@RequestParam(value = "imie", required = true) String firstName,
+                              @RequestParam(value = "nazwisko", required = true) String lastName,
+                              @RequestParam(value = "wiek", required = true) String ageStr) {
+        // Sprawdzenie, czy imię, nazwisko są niepuste i czy wiek jest poprawny
+        if (firstName.isEmpty() || firstName.equalsIgnoreCase("null") ||
+                lastName.isEmpty() || lastName.equalsIgnoreCase("null") ||
+                ageStr.isEmpty() || ageStr.equalsIgnoreCase("null")) {
+            return "redirect:/err.html";  // Przekierowanie na stronę błędu, jeśli dane są nieprawidłowe
+        }
+
+        try {
+            int age = Integer.parseInt(ageStr);  // Próbujemy zamienić wiek na liczbę całkowitą
+            if (age < 0) {  // Sprawdzamy, czy wiek jest nieujemny
+                return "redirect:/err.html";
+            }
+            userRepository.addUser(new User(firstName, lastName, age));
+            return "redirect:/success.html";  // Przekierowanie na stronę sukcesu
+        } catch (NumberFormatException e) {
+            return "redirect:/err.html";  // Przekierowanie na stronę błędu w przypadku niepoprawnego formatu wieku
```

Comment:
Zamiast `Integer.parseInt(ageStr)` możesz od razu wskazać, że wiek to integer:

`@RequestParam(value = "wiek") int age`

Reply:
owszem, ale jak miałem to jako zwykły int, to w momencie, kiedy ktoś uparcie doda "wiek" w adresie jako "abc", albo "null", to jesteśmy niechronieni

Reply:
to specjalnie zostało najpierw opakowane w Stringa, żeby właśnie można było sprawdzić `ageStr.isEmpty() || ageStr.equalsIgnoreCase("null")`
na tyle nie ufam użytkownikom, że nawet użyłem `ignoreCase()` 😄

Reply:
Co masz na myśli, "niechronieni"? Powinna zostać wyświetlona strona z błędem i to się raczej dzieje. W teorii najgorsza opcja to dodać do bazy danych dane niespójne, czyli np. wiek to `null`. 

Teraz zauważyłem: `firstName.equalsIgnoreCase("null")` - tutaj porównujesz wartość do Stringa o wartości "null" - a nie do `null`, czyli nieustawionej referencji. Wydaje mi się, że powinno być to drugie. A jeszcze lepiej  `org.springframework.util.StringUtils.hasText("abc");`, które wygląda tak:

```
	public static boolean hasText(@Nullable String str) {
		return (str != null && !str.isBlank());
	}
```

Reply:
to wszystko wzięło się z tego, że formularz jest zabezpieczony przez `required` - nie da się wysłać ze strony głównej formularza bez podania imienia i nazwiska (składających się ze Stringa) i wieku (int). To mamy zabezpieczone.
Ale jednocześnie wpisując np. `/add?imie=null&nazwisko=Abacki&wiek=15` to dla HTML-a "null" nie będzie pustym obiektem a ciągiem znaków "null". Więc po to porównanie do Stringa.
Przed "obiektowym nullem" bronię się przez `firstName.isEmpty()`.

Nie chodzi mi o 0,2 pkt, zwłaszcza, że mam inne, drobne, błędy, ale po prostu tłumaczę, dlaczego zrobiłem tak, a nie inaczej, bronię pomysłu. 😄

Reply:
analogicznie było z wiekiem - w formularzu nie da się wpisać wieku jako "-15", albo "abc", ale w pasku adresu już owszem.